rbd/snapshot_namespace.go

```diff
@@ -0,0 +1,39 @@
+// +build !luminous
+//
+// Ceph Mimic introduced rbd_snap_get_namespace_type().
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <rbd/librbd.h>
+import "C"
+
+// SnapNamespaceType indicates the namespace to which the snapshot belongs to.
+type SnapNamespaceType C.rbd_snap_namespace_type_t
+
+const (
+	// SnapNamespaceTypeUser indicates that the snapshot belongs to user namespace.
+	SnapNamespaceTypeUser = SnapNamespaceType(C.RBD_SNAP_NAMESPACE_TYPE_USER)
+
+	// SnapNamespaceTypeGroup indicates that the snapshot belongs to group namespace.
+	// Such snapshots will have associated group information.
+	SnapNamespaceTypeGroup = SnapNamespaceType(C.RBD_SNAP_NAMESPACE_TYPE_GROUP)
+
+	// SnapNamespaceTypeTrash indicates that the snapshot belongs to trash namespace.
+	SnapNamespaceTypeTrash = SnapNamespaceType(C.RBD_SNAP_NAMESPACE_TYPE_TRASH)
+)
+
+// GetSnapNamespaceType gets the type of namespace to which the snapshot belongs to,
+// returns error on failure.
+// Implements:
+//  int rbd_snap_get_namespace_type(rbd_image_t image, uint64_t snap_id, rbd_snap_namespace_type_t *namespace_type)
+func (image *Image) GetSnapNamespaceType(snapID uint64, namespace *SnapNamespaceType) error {
```

Comment:
The namespace value should be returned from the function not a pointer.

Reply:
The reason I didn't return the namespace but decided to pass a pointer so that the value can be set is because I was not sure what to return in case of a failure.
Basically, lets say rbd_snap_get_namespace_type() returns error and I return **x, getError(ret)**. Now this 'x' needs to be of type rbd_snap_namespace_type_t which is not possible AFAIT.

Reply:
Go convention would be to have it be the zero value for that type. Simply doing:
```
var nstype SnapNamespaceType
ret := C.rbd_snap_get_namespace_type(image.image,
		C.uint64_t(snapID),
		(*C.rbd_snap_namespace_type_t)(&nstype))
	return nstype, getError(ret)
```
should do the right thing.

Reply:
Yeah, I initially thought of it and it will do the right thing but even in failure cases it will set nstype value as 0 which is still a legitimate value. Also, it looks a bit ugly to me, I have made the changes PTAL.